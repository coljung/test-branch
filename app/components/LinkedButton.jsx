import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button } from 'antd';
import { browserHistory } from 'react-router';

export default class LinkedButton extends Component {
    navigate(to) {
        browserHistory.push(to);
    }

    render() {
        return (
            <Button onClick={this.navigate.bind(this, this.props.to)} {...this.props}>{this.props.children}</Button>
        );
    }
}

LinkedButton.propTypes = {
    to: PropTypes.string.isRequired,
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node,
    ]).isRequired,
};
