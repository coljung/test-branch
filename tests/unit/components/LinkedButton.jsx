import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Button } from 'antd';
import LinkedButton from '../../../app/components/LinkedButton';
import { browserHistory } from 'react-router';

// Mock react-router lib
jest.mock('react-router');

describe('LinkedButton', () => {

    it('should render correctly', () => {
        const output = shallow(
            <LinkedButton title="mockTitle" to="mockUrl">Foo Bar</LinkedButton>
        );

        expect(output.find('Button').length).toBe(1);

        output.find('Button').simulate('click');

        expect(browserHistory.push).toHaveBeenCalledTimes(1);
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});
