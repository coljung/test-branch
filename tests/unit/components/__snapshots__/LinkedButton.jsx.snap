// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkedButton should render correctly 1`] = `
<Button
  ghost={false}
  loading={false}
  onClick={[Function]}
  prefixCls="ant-btn"
  title="mockTitle"
  to="mockUrl"
>
  Foo Bar
</Button>
`;
